moped-editor/src/utils/mapHelpers.js

```diff
@@ -296,8 +298,16 @@ export const createSummaryMapLayers = (selectedIds, geoJSON) => {
           {...createProjectViewLayerConfig(sourceLayerName)}
         />
       </Source>
-    )
-  );
+    ))
+    .sort((a, b) => {
+      // The id of the Source component maps to the source layer names in mapConfig, each layer config has a set order
+      const idA = a.props.id;
+      const idB = b.props.id;
+      const orderA = mapConfig.layerConfigs[idA].layerOrder;
+      const orderB = mapConfig.layerConfigs[idB].layerOrder;
+
+      return orderA > orderB ? 1 : -1;
+    });
```

Comment:
Daang, it sounded a lot easier than this. The fact that you have to sort by layer order... 🙏

Reply:
You probably already know it, but a thought crossed my wires. Is there anything [z-index](https://developer.mozilla.org/en-US/docs/Web/CSS/z-index) can do to save us? A long shot in my head but worth asking,

Reply:
You and I are on the same page! The documented way to manage this is [beforeId](https://visgl.github.io/react-map-gl/docs/api-reference/layer#beforeid) which is pretty much like z-index for Mapbox layers. It gets a little confusing and hard to define the order because the argument is a layer id attribute, but your feedback helped me go that direction. It shouldn't be hard to refactor since there are only a few layers, and I like it a lot better! 🙏